openbas-api/src/main/java/io/openbas/rest/helper/RestBehavior.java

```diff
@@ -206,4 +206,12 @@ public void checkOrganizationAccess(UserRepository userRepository, String organi
       }
     }
   }
+
+  protected void validateUUID(final String id) {
+    try {
+      UUID.fromString(id);
+    } catch (IllegalArgumentException e) {
+      throw new IllegalArgumentException("ID invalid : " + id, e);
```

Comment:
Will this cause HTTP 500, where it should be HTTP 400 IMO?

Reply:
How about throwing https://www.baeldung.com/spring-response-status-exception instead, wrapping the original exception?

Reply:
You are right
I can throwing an ElementNotFoundException and it will be handled by the RestBehavior class

  @ExceptionHandler(ElementNotFoundException.class)
  public ResponseEntity<ErrorMessage> handleElementNotFoundException(ElementNotFoundException ex) {
    ErrorMessage message = new ErrorMessage("Element not found: " + ex.getMessage());
    log.warning("ElementNotFoundException: " + ex.getMessage());
    return new ResponseEntity<>(message, HttpStatus.NOT_FOUND);
  }

Reply:
That's for HTTP 404, but perhaps there is a readymade handler for 400 as well ?N

Reply:
I can use InputValidationException

  @ResponseStatus(HttpStatus.BAD_REQUEST)
  @ExceptionHandler(InputValidationException.class)
  public ValidationErrorBag handleInputValidationExceptions(InputValidationException ex) {
    ValidationErrorBag bag = new ValidationErrorBag();
    ValidationError errors = new ValidationError();
    Map<String, ValidationContent> errorsBag = new HashMap<>();
    errorsBag.put(ex.getField(), new ValidationContent(ex.getMessage()));
    errors.setChildren(errorsBag);
    bag.setErrors(errors);
    return bag;
  }

Reply:
sounds like a plan

Reply:
I made the changes.